modules/post/windows/gather/ad_to_sqlite.rb

```diff
@@ -377,7 +377,7 @@ def run_sqlite_query(db, table_name, values)
   def create_sqlite_db
     begin
       obj_temp = ::Dir::Tmpname
```

Comment:
Should this be in loot rather than tmp? 🤔

Reply:
For me, if this is left as tmp it doesn't work 👀 
```
msf6 post(windows/gather/ad_to_sqlite) > rerun session=-1
[*] Reloading module...

[-] Error(Database): unable to open database file
[*] Database created: 
[-] Post failed: NoMethodError undefined method `execute' for nil:NilClass
[-] Call stack:
[-]   /Users/user/Documents/code/metasploit-framework/modules/post/windows/gather/ad_to_sqlite.rb:373:in `run_sqlite_query'
[-]   /Users/user/Documents/code/metasploit-framework/modules/post/windows/gather/ad_to_sqlite.rb:141:in `block (2 levels) in run'
[-]   /Users/user/Documents/code/metasploit-framework/lib/msf/core/thread_manager.rb:105:in `block in spawn'
[-]   /Users/user/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/logging-2.3.0/lib/logging/diagnostic_context.rb:474:in `block in create_with_logging_context'
[*] Post module execution completed
```

Reply:
A quick hack shows if it were saved to the msf4 loot directory it would work fine:

```ruby
      # obj_temp = ::Dir::Tmpname
      # filename = "#{obj_temp.tmpdir}/#{obj_temp.create('ad_',max_tries: 2) {|e| e}}.db"
      if ! ::File.directory?(::Msf::Config.loot_directory)
        FileUtils.mkdir_p(::Msf::Config.loot_directory)
      end
      filename = ::File.join(::Msf::Config.loot_directory, "ad_test.db")
      db = SQLite3::Database.new(filename)
      db.type_translation = true
```

Result:

```
msf6 post(windows/gather/ad_to_sqlite) > rerun session=-1
[*] Reloading module...

[*] Database created: /Users/user/.msf4/loot/ad_test.db
[*] Database closed: /Users/user/.msf4/loot/ad_test.db at 73728 byte(s)
[*] Post module execution completed
```

Obviously just a quick hack, and not the recommended code to use 😄

Reply:
https://github.com/sempervictus/metasploit-framework/pull/32